app/src/components/features/rules/TestThisRule/TestThisRule.tsx

```diff
@@ -0,0 +1,175 @@
+          };
+          const sessionEvents = tabSession.events;
+
+          const recordingOptionsToSave = getRecordingOptionsToSave([
+            DebugInfo.INCLUDE_CONSOLE_LOGS,
+            DebugInfo.INCLUDE_NETWORK_LOGS,
+          ]);
+
+          saveRecording(
+            user.details?.profile?.uid,
+            workspace?.id,
+            sessionMetadata,
+            compressEvents(getSessionEventsToSave(sessionEvents, recordingOptionsToSave)),
+            recordingOptionsToSave,
+            SOURCE.TEST_THIS_RULE,
+            true
+          ).then((response) => {
+            if (response.success) {
+              trackTestRuleSessionDraftSaved(SessionSaveMode.ONLINE);
+              getTestReportById(appMode, reportId).then((testReport) => {
+                if (testReport) {
+                  testReport.sessionLink = getSessionRecordingSharedLink(response?.firestoreId);
+                  saveTestReport(appMode, reportId, testReport).then(() => {
+                    fetchAndUpdateTestReports();
+                  });
+                }
+              });
+            }
+          });
+        })
+        .catch((error) => {
+          Logger.log(error);
+          toast.error("Error saving test session");
+        });
+    },
+    [appMode, user.details?.profile?.uid, workspace?.id, fetchAndUpdateTestReports]
+  );
+
+  const handleTestReportDelete = useCallback(
+    (reportId: string) => {
+      deleteTestReport(appMode, reportId)
+        .then(() => {
+          toast.success("Test deleted successfully");
+          trackTestRuleReportDeleted(currentlySelectedRuleData.ruleType);
+          fetchAndUpdateTestReports();
+        })
+        .catch((error) => {
+          Logger.log(error);
+        });
+    },
+    [appMode, currentlySelectedRuleData.ruleType, fetchAndUpdateTestReports]
+  );
+
+  useEffect(() => {
+    PageScriptMessageHandler.addMessageListener(
+      GLOBAL_CONSTANTS.EXTENSION_MESSAGES.NOTIFY_TEST_RULE_REPORT_UPDATED,
+      (message: { testReportId: string; testPageTabId: string; record: boolean; appliedStatus: boolean }) => {
+        fetchAndUpdateTestReports(message.testReportId);
+        trackTestRuleReportGenerated(currentlySelectedRuleData.ruleType, message.appliedStatus);
+        if (isSheetPlacedAtBottom) {
+          openBottomSheet();
+        }
+        if (message.record) {
+          handleSaveTestSession(parseInt(message.testPageTabId), message.testReportId);
+        }
+      }
+    );
+  }, [
+    currentlySelectedRuleData.ruleType,
+    handleSaveTestSession,
+    fetchAndUpdateTestReports,
+    isBottomSheetOpen,
+    openBottomSheet,
+    isSheetPlacedAtBottom,
+  ]);
+
+  useEffect(() => {
+    fetchAndUpdateTestReports();
+  }, [fetchAndUpdateTestReports]);
+
+  useEffect(() => {
+    // Open the bottom sheet when a new rule is created
+    if (isNewRuleCreated.current) {
+      openBottomSheet();
+      isNewRuleCreated.current = false;
+    }
```

Comment:
RuleEditor should open bottomsheet when new rule is created. BottomSheet shouldn't listen for rule creation event and open itself. This will keep all the actions triggered from rule creation in one place